frontend/src/hooks/useDarkMode.tsx

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from "react";
+
+const useDarkMode = (): [boolean, () => void] => {
+   const [darkMode, setDarkMode] = useState<boolean>(() => {
+      const savedMode = localStorage.getItem("darkMode");
+      return savedMode ? JSON.parse(savedMode) : false;
```

Comment:
put this out of use-state on its own variable, and also either don't use JSON or if you do handle any errors by falling back to the default theme